SPEC.md

```diff
@@ -7367,6 +7368,93 @@ Example output:
 </p>
 </details>
 
+### ✨ `join_paths`
+
+```
+File join_paths(File|String, String)
+File join_paths(File, Array[String]+)
+File join_paths(Array[String]+)
+```
+
+Joins together two or more paths into an absolute path in the host filesystem.
+
+There are three variants of this function:
+
+1. `File join_paths(File|String, String)`: Joins together exactly two paths. The first path may be either absolute or relative and must specify a directory; the second path is relative to the first path and may specify a file or directory.
+2. `File join_paths(File, Array[String]+)`: Joins together any number of relative paths with a base path. The first argument may be either an absolute or a relative path and must specify a directory. The paths in the second array argument must all be relative. The *last* element may specify a file or directory; all other elements must specify a directory.
+3. `File join_paths(Array[String]+)`: Joins together any number of paths. The array must not be empty. The *first* element of the array may be either absolute or relative; subsequent path(s) must be relative. The *last* element may specify a file or directory; all other elements must specify a directory.
+
+An absolute path starts with `/` and indicates that the path is relative to the root of the container in which the task is executed. A relative path does not start with `/` and indicates the path is relative to its parent directory. It is up to the execution engine to determine which directory to use as the parent when resolving relative paths; by default it is the working directory in which the task is executed.
+
+**Parameters**
+
+1. `File|String|Array[String]+`: Either a path or an array of paths.
+2. `String|Array[String]+`: A relative path or paths; only allowed if the first argument is not an array.
+
+**Returns**: A `File` representing an absolute path that results from joining all the paths in order (left-to-right), and resolving the resulting path against the default parent directory if it is relative.
+
+<details>
+<summary>
+Example: join_paths_task.wdl
+
+```wdl
+version 1.2
+
+task resolve_paths_task {
+  input {
+    File abs_file = "/usr"
+    String abs_str = "/usr"
+    String rel_dir_str = "bin"
+    File rel_file = "echo"
+    File rel_dir_file = "mydir"
+    String rel_str = "mydata.txt"
+  }
+
+  # these are all equivalent to '/usr/bin/echo'
+  File bin1 = join_paths(abs_file, [rel_str_dir, rel_file])
+  File bin2 = join_paths(abs_str, [rel_str_dir, rel_file])
+  File bin3 = join_paths([abs_str, rel_str_dir, rel_file])
+  
+  # this resolves to '<working dir>/mydir/mydata.txt'
```

Comment:
Depends how pedantic the spec should be, but the above states:

> It is up to the execution engine to determine which directory to use as the parent when resolving relative paths;

So maybe "the default behaviour is this should resolve to ..."?